source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOAdaptiveCard.mm

```diff
@@ -56,40 +56,115 @@ - (NSData *)inputs
+    NSError *regexError = nil;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"(\\\\+\\.)"
+                                                                           options:0
+                                                                             error:&regexError];
+    if (regexError != nil) {
+        NSLog(@"Regex Error: %@", regexError.localizedDescription);
+        return payload; // Return the original payload if regex creation fails
+    }
+    
+    NSMutableString *mutablePayload = [payload mutableCopy];
+
+    // Replace matches with the correct number of escaped backslashes
+    [regex enumerateMatchesInString:payload
+                            options:0
+                              range:NSMakeRange(0, payload.length)
+                         usingBlock:^(NSTextCheckingResult *match, NSMatchingFlags flags, BOOL *stop) {
+        if (match.range.length > 0) {
+            NSString *matchedString = [payload substringWithRange:match.range];
+            NSString *replacementString = [matchedString stringByReplacingOccurrencesOfString:@"\\" withString:@"\\\\"];
+            [mutablePayload replaceCharactersInRange:match.range withString:replacementString];
+        }
+    }];
+
+    return [mutablePayload copy];
+}
+
++ (BOOL)isValidJson:(NSString *)jsonString error:(NSError **)error {
+    NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+    if (!jsonData) {
+        return NO;
+    }
+
+    [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:error];
+    return (*error == nil);
+}
+
+
++ (ACOAdaptiveCardParseResult *)fromJson:(NSString *)payload {
     const std::string g_version = "1.6";
     ACOAdaptiveCardParseResult *result = nil;
     if (payload) {
-        try {
-            ACOAdaptiveCard *card = [[ACOAdaptiveCard alloc] init];
-            std::shared_ptr<ParseResult> parseResult = AdaptiveCard::DeserializeFromString(std::string([payload UTF8String]), g_version);
-            NSMutableArray *acrParseWarnings = [[NSMutableArray alloc] init];
-            std::vector<std::shared_ptr<AdaptiveCardParseWarning>> parseWarnings = parseResult->GetWarnings();
-            for (const auto &warning : parseWarnings) {
-                ACRParseWarning *acrParseWarning = [[ACRParseWarning alloc] initWithParseWarning:warning];
-                [acrParseWarnings addObject:acrParseWarning];
+        NSError *jsonError = nil;
+
+        // First, check if the JSON is valid
+        if (![self isValidJson:payload error:&jsonError]) {
+            NSLog(@"Initial JSON Deserialization Error: %@", jsonError.localizedDescription);
+            jsonError = nil;
+
+            // Attempt to fix the JSON using regex replacement
+            NSString *processedPayload = [self correctInvalidJsonEscapes:payload];
+
+            // Check if the corrected JSON is valid
+            if (![self isValidJson:processedPayload error:&jsonError]) {
```

Comment:
Why do we need to check again after post processing? Does this mean that the correctInvalidJsonEscapes method could potentially create invalid jsons?

Reply:
In regards to @GabrielMedAlv  concern about increasing parsing times for AC in general, I circumvent this by only doing this check if we detect an error in the original json payload. The intended flow is:

- Check if the AC json is valid
- if the AC json isn't valid, attempt to fix it by fixing faulty Regex cases
- Validate fixed json
- continue as normal

before this the flow was:
- If the AC Json isn't valid, do nothing and card is broken

Reply:
RE: In regards to performance the new intended flow is:

- Check if the AC json is valid
- if the AC json isn't valid, attempt to fix it by fixing faulty Regex cases
- continue as normal

(we don't need to validate the fixed json, it either works now or fails how we had it before)